esrally/track/loader.py

```diff
@@ -1115,6 +1135,24 @@ def meta_data(self):
         }
 
 
+class TrackDependencyRegistry:
+    """
+    Installs track-specific dependencies to the local runtime
+    """
+
+    def __init__(self, install_dependencies):
+        self.install_dependencies = install_dependencies
+
+    def __call__(self, name):
+        if self.install_dependencies:
+            console.info(f"Installing track dependency {name}")
+            try:
+                with open(os.path.join(paths.logs(), "dependency.log"), "ab") as install_log:
+                    subprocess.check_call([sys.executable, "-m", "pip", "install", f"{name}"], stdout=install_log, stderr=install_log)
```

Comment:
One problem I see with this way of installation, is that we support multiple ways of installing Rally:

While for development environments we expect most people to use a venv, [we document](https://esrally.readthedocs.io/en/latest/install.html#installing-rally) (and have uses of, e.g. in nightly envs) installing Rally using the [user mode scheme](https://docs.python.org/3/install/index.html#alternate-installation-the-user-scheme). These [user installs](https://pip.pypa.io/en/stable/user_guide/#user-installs), are done using the `--user` flag.

Doing a `python3 -mpip install` here will not honor any user setups and in those cases end up installing the package in unexpected locations.

I don't think it's a big problem determining if `esrally` has been installed with `--user`; we could leverage the information in https://docs.python.org/3/library/site.html#site.USER_SITE or maybe there are even simpler ways.

Reply:
IMO supportability concerns such as you outline here (and worse, the ones we aren't anticipating), as well as--for lack of a better word--stupidity concerns that might fundamentally change how this feature is implemented and consumed after trying it in the real world are both the reasons I'd go the extreme `TrackPreparator` route and leave this undocumented.

@pquentin I think your point is valid, indeed this is exactly what an experimental feature designation is for, but there's a precedent for the *really* shaky (AKA evolving) stuff that keeps us from even documenting how to use it until it's had some time to harden and evolve. If this is not compelling to you, though, I'll go the documented-but-unsupported route as you previously suggested

Reply:
I *think* I understand your concerns, but I still don't agree, sorry. :( In Rally, I'd say documentation is as import as tests. Would you say "this is experimental, let's skip the tests"?

It's also simple to add, you only need a short new section at the bottom of docs/adding_tracks.rst with a warning plus a code block with the call to `register.register_dependency` you have in the tests. This should not prevent us from fearlessly updating/removing this feature.

We can ask others to act as tie breakers if you'd prefer?

Reply:
Will write up the feature, and tag it as experimental.  Thanks